zerver/tests/test_push_registration.py

```diff
@@ -190,3 +197,192 @@ def test_register_push_device_error(self) -> None:
             invalid_token_payload,
         )
         self.assert_json_error(result, "Invalid encrypted_push_registration")
+
+
+class RegisterPushDeviceToServer(ZulipTestCase):
+    @staticmethod
+    def get_register_push_device_payload() -> dict[str, str | int]:
+        payload: dict[str, str | int] = {
+            "token_kind": PushDevice.TokenKind.FCM,
+            "push_account_id": 2408,
+            "push_public_key": "dummy-push-public-key",
+            "bouncer_public_key": "bouncer-public-key",
+            "encrypted_push_registration": "encrypted-push-registration",
+        }
+        return payload
+
+    @mock.patch(
```

Comment:
Maybe put a comment on this explaining the strategy for mocking here. But I think I'd prefer to see this test just being properly end-to-end rather than mocking that? It shouldn't be hard to just do the encryption for these tests.

Reply:
Reworked the test to test things end-to-end as discussed.